pre-processing/primod/ribamod/ribamod.py

```diff
@@ -184,28 +188,27 @@ def validate_keys(
                 f"present in the model: {missing}"
             )
 
-    @staticmethod
-    def derive_river_drainage_coupling(
-        gridded_basin: xr.DataArray,
-        basin_ids: "pd.Series[int]",
-        conductance: xr.DataArray,
-    ) -> pd.DataFrame:
-        # Conductance is leading parameter to define location, for both river
-        # and drainage.
-        # FUTURE: check for time dimension? Also order and inclusion of layer
-        # in conductance.
-        # Use xarray where to force the dimension order of conductance.
-        basin_id = xr.where(conductance.notnull(), gridded_basin, np.nan)  # type: ignore
-        include = basin_id.notnull().to_numpy()
-        basin_id_values = basin_id.to_numpy()[include].astype(int)
-        # Ribasim internally sorts the basin, which determines the order of the
-        # Ribasim level array.
-        basin_index = np.searchsorted(basin_ids, basin_id_values)
-        boundary_index_values = np.cumsum(conductance.notnull().to_numpy().ravel()) - 1
-        boundary_index_values = boundary_index_values[include.ravel()]
-        return pd.DataFrame(
-            data={"basin_index": basin_index, "bound_index": boundary_index_values}
-        )
+    def validate_basin_node_ids(self) -> "pd.Series[int]":
+        assert self.ribasim_model.basin.profile.df is not None
+        basin_ids = np.unique(self.ribasim_model.basin.profile.df["node_id"])
+        missing = ~np.isin(self.basin_definition["node_id"], basin_ids)
+        if missing.any():
+            missing_basins = self.basin_definition["node_id"].to_numpy()[missing]
+            raise ValueError(
+                "The node IDs of these basins in the basin definition do not "
+                f"occur in the Ribasim model: {missing_basins}"
+            )
+        return basin_ids
+
+    def validate_subgrid_df(self, coupling: DriverCoupling) -> pd.DataFrame | None:
```

Comment:
Does it make sense here to return `None` instead of raising an exception

Reply:
An exception should definitely be raised if you're trying to actively couple, but you forgot to define the subgrid thing.